docs/src/service-workers-guide.md

```diff
@@ -0,0 +1,161 @@
+      if (response) return response;
+
+      // 2. Re-write request for /foo to /bar
+      if (event.request.url.endsWith("foo")) return fetch("./bar");
+
+      // 3. Prevent `tracker.js` from being retrieved, and returns a placeholder response
+      if (event.request.url.endsWith("tracker.js"))
+        return new Response('conosole.log("no trackers!")', {
+          status: 200,
+          headers: { "Content-Type": "text/javascript" },
+        });
+
+      // 4. Otherwise, fallthrough, perform the fetch and respond
+      return fetch(event.request);
+    })()
+  );
+});
+
+self.addEventListener("activate", (event) => {
+  event.waitUntil(clients.claim());
+});
+```
+
+And a Page that simply registers the Service Worker:
+
+```html
+<!-- filename: index.html -->
+<script>
+  window.registrationPromise = navigator.serviceWorker.register('./service-worker-main.js');
+</script>
+```
+
+On the first visit to the page via [`method: Page.goto`], the following Request/Response events would be emitted:
+
+| Scope   | Target         | URL                       | Routed | fromServiceWorker |
+| -       | -              | -                         | -      | -                 |
+| Context | Frame          | index.html                | Yes    |                   |
+| Page    | Frame          | index.html                | Yes    |                   |
+| Context | Service Worker | service-worker-main.js    | Yes    |                   |
+| Context | Service Worker | addressbook.json          | Yes    |                   |
+
+
+Once the Service Worker is activated and handling FetchEvents, if the page makes the following requests:
+
+```js
+await page.evaluate(() => fetch('./addressbook.json'));
+await page.evaluate(() => fetch('./foo'));
+await page.evaluate(() => fetch('./tracker.js'));
+await page.evaluate(() => fetch('./fallthrough.txt'));
+```
+
+The following Request/Response events would be emitted:
+
+| Scope   | Target         | URL                       | Routed | fromServiceWorker |
+| -       | -              | -                         | -      | -                 |
+| Context | Frame          | addressbook.json          |        | Yes               |
+| Page    | Frame          | addressbook.json          |        | Yes               |
+| Context | Service Worker | bar                       | Yes    |                   |
+| Context | Frame          | foo                       |        | Yes               |
+| Page    | Frame          | foo                       |        | Yes               |
+| Context | Frame          | tracker.js                |        | Yes               |
+| Page    | Frame          | tracker.js                |        | Yes               |
+| Context | Service Worker | fallthrough.txt           | Yes    |                   |
+| Context | Frame          | fallthrough.txt           |        | Yes               |
+| Page    | Frame          | fallthrough.txt           |        | Yes               |
```

Comment:
I think this table makes it much clearer (and it's generate from a Test nearly verbatim):

<img width="760" alt="Screen Shot 2022-07-13 at 2 12 17 AM" src="https://user-images.githubusercontent.com/11915034/178699293-8c4d08d0-ad43-4253-a146-90325deda97d.png">